lib/ansible/module_utils/facts/system/ssh_pub_keys.py

```diff
@@ -47,6 +47,10 @@ def collect(self, module=None, collected_facts=None):
                 key_filename = '%s/ssh_host_%s_key.pub' % (keydir, type_)
                 keydata = get_file_content(key_filename)
                 if keydata is not None:
-                    ssh_pub_key_facts[factname] = keydata.split()[1]
+                    (subtype, key) = keydata.split()
```

Comment:
Actually, it's a bit more complicated even than this. From the `sshd` manpage on Fedora 25:
```
Public keys consist of the following space-separated fields: options, keytype, base64-encoded
     key, comment.  The options field is optional.  The keytype is “ecdsa-sha2-nistp256”, “ecdsa-sha2-nistp384”, “ecdsa-sha2-nistp521”, “ssh-ed25519”, “ssh-dss” or “ssh-rsa”; the comment field is not used
     for anything (but may be convenient for the user to identify the key).
```